client/state/happychat/user/actions.js

```diff
@@ -0,0 +1,30 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import wpcom from 'lib/wp';
+import { HAPPYCHAT_ELIGIBILITY_SET } from 'state/action-types';
+import { errorNotice } from 'state/notices/actions';
+
+export const setHappyChatEligibility = isEligible => ( {
+	type: HAPPYCHAT_ELIGIBILITY_SET,
+	isEligible,
+} );
+
+export const requestHappychatEligibility = () => dispatch => {
```

Comment:
For context: there is the [happychat-client](https://github.com/Automattic/happychat-client/) project which is going to be the source of truth for Happychat customer facing code. In a few weeks, Calypso will be modified to use this library and the only code remaining will be Calypso-specific.

In that context, I've been decoupling Calypso-specific code into [middleware-calypso](https://github.com/Automattic/wp-calypso/blob/master/client/state/happychat/middleware-calypso.js), all the rest should be Calypso-agnostic. Can this be refactored for the Olark request to be fired on middleware-calypso?

The `HAPPYCHAT_USER_ELIGIBILITY` actions are still valuable because this is something we'll need in every host, so I'll take care of merging these actions in the consolidated `state.user` for [happychat-client](https://github.com/Automattic/happychat-client/tree/master/src/state/user).

Reply:
Follow-up: the `requestChatEligibility` could be either a Happychat or a host (Calypso, WordPress, etc) action. I'll probably lean towards that being a host action, but I haven't thought about it deeply yet, so use whatever works for you at the moment - I can refactor that later if needed.